test/integration/QrochetQuacExt_test.jl

```diff
@@ -20,4 +20,67 @@
         notsiteinds = setdiff(inds(TensorNetwork(qftqtn)), siteinds)
         @test_skip issetequal(inds(TensorNetwork(qftqtn), :inner), notsiteinds)
     end
+
+    @testset "evolve" begin
+        n = 10
+        timesteps = 20
+        δₜ = 0.1
+        ket₊ = 1 / √2 * [1, 1]
+        observables = Dense.([Z(5)])
+
+        function trotter_XX(i, j; δₜ = δₜ)
+            mat = kron(Matrix(X()), Matrix(X()))
+            mat = cis(δₜ * mat)
+            mat = reshape(mat, 2, 2, 2, 2)
+            Dense(Qrochet.Operator(), mat; sites = [Site(i), Site(j), Site(i, dual = true), Site(j, dual = true)])
+        end
+
+        function trotter_Z(i; λ, δₜ = δₜ)
+            mat = Matrix(Z())
+            mat = cis(-λ * δₜ * mat)
+
+            Dense(Qrochet.Operator(), mat; sites = [Site(i), Site(i, dual = true)])
+        end
+
+        @testset "not canonical" begin
+            ψ = convert(Chain, Product(fill(ket₊, n)))
+
+            for it in 1:timesteps
+                for (i, j) in Iterators.filter(==(2) ∘ length, Iterators.partition(1:n, 2))
+                    evolve!(ψ, trotter_XX(i, j; δₜ))
+                end
+                for (i, j) in Iterators.filter(==(2) ∘ length, Iterators.partition(2:n, 2))
+                    evolve!(ψ, trotter_XX(i, j; δₜ))
+                end
+                for i in 1:n
+                    evolve!(ψ, trotter_Z(i; λ = 0.3, δₜ))
+                end
+            end
+        end
```

Comment:
I don't think this is the best way to test `evolve!`. Also, you are not testing anything here (there is no `@test`).

Tests should be much smaller and fundamental.

Reply:
I would just remove theses tests and write tests for `evolve!` in another PR.

Reply:
Okey, let's do this. Anyway, we should have some tests! It scares me to PR without testing 🥲